Build failure: arma3-unix-launcher

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `arma3-unix-launcher`

### Build log

```
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/553i55vjpf89bqj5dc3fkjp1y971rs76-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/2ki1czqqy7a47c39wgc07d4bv1ikabfh-dont_fetch_dependencies.patch
patching file cmake/external_dependencies.cmake
applying patch /nix/store/c1a513bw9xscij4n2243jdnb5yx6r592-disable_steam_integration.patch
patching file src/arma3-unix-launcher/mainwindow.cpp
patching file src/dayz-linux-launcher/mainwindow.cpp
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/8f197yxrxjwr
9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-ar
ma3-unix-launcher-413/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/share/doc/arma3-unix-launcher -DCMAKE_INSTALL_INFODIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-laun
cher-413/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/include -DC
MAKE_INSTALL_INCLUDEDIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/include -DCMAKE_INSTALL_SBINDIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/sbin -DCMAKE_INSTALL_BINDIR=/nix/store
/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_
STRIP=/nix/store/hgw57m0jh2w10jsv8x86ng267pm5sla8-gcc-wrapper-13.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/hgw57m0jh2w10jsv8x86ng267pm5sla8-gcc-wrapper-13.3.0/bin/ranlib -DCMAKE_AR=/nix/store/hgw57m0jh2w10jsv8x86ng267pm5sla8-gcc-wra
pper-13.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/8f197yxrxjwr9li04dxcrp74iwc2vhvz-arma3-unix-launcher-413 -Wno-dev
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/hgw57m0jh2w10jsv8x86ng267pm5sla8-gcc-wrapper-13.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/hgw57m0jh2w10jsv8x86ng267pm5sla8-gcc-wrapper-13.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test SYSTEM_argparse_WORKS
-- Performing Test SYSTEM_argparse_WORKS - Failed
-- Downloading argparse from /nix/store/qrxph6ygycbzp0cp7sijcvzhdk7c7q7x-source
-- Using external argparse
-- Performing Test SYSTEM_curlpp_WORKS
-- Performing Test SYSTEM_curlpp_WORKS - Failed
-- Downloading curlpp from /nix/store/cxisn2rips8bkizbvzfz6y11ykdcyckp-source
-- curlpp version=[0.8.1]
-- curlpp version num=[000801]
-- Looking for CURL
-- Found CURL: /nix/store/vic96jkyhfvq58bwch08618zg72gdgkz-curl-8.10.1/lib/libcurl.so (found version "8.10.1")
-- Found CURL version: 8.10.1
-- Using CURL include dir(s): /nix/store/h9ff1kzr3xfijmxqgxjbbmjac29sh9nj-curl-8.10.1-dev/include
-- Using CURL lib(s): /nix/store/vic96jkyhfvq58bwch08618zg72gdgkz-curl-8.10.1/lib/libcurl.so
-- Using external curlpp
-- Performing Test SYSTEM_fmt_WORKS
-- Performing Test SYSTEM_fmt_WORKS - Success
-- Performing Test SYSTEM_pugixml_WORKS
-- Performing Test SYSTEM_pugixml_WORKS - Failed
-- Downloading pugixml from /nix/store/25biwwg7sxh6j9g8rrpharzfj1izl0ig-source
-- Using external pugixml
-- Performing Test SYSTEM_nlohmann_json_WORKS
-- Performing Test SYSTEM_nlohmann_json_WORKS - Failed
-- Performing Test SYSTEM_spdlog_WORKS
-- Performing Test SYSTEM_spdlog_WORKS - Success
-- Configuring done (5.5s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j32 SHELL=/nix/store/mm0pa3z7kk6jh1i9rkxqxjqmd8h1qpxf-bash-5.2p37/bin/bash
[  1%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Exception.cpp.o
[  3%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Form.cpp.o
[  5%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Info.cpp.o
[  8%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Easy.cpp.o
[  8%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Multi.cpp.o
[ 10%] Building CXX object _deps/pugixml-build/CMakeFiles/pugixml.dir/src/pugixml.cpp.o
[ 11%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/OptionBase.cpp.o
[ 13%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Options.cpp.o
[ 15%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/internal/CurlHandle.cpp.o
[ 16%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/cURLpp.cpp.o
[ 18%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/internal/OptionList.cpp.o
[ 20%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/internal/OptionSetter.cpp.o
[ 22%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/internal/SList.cpp.o
In file included from /build/source/build/_deps/curlpp-src/src/curlpp/Options.cpp:25:
/build/source/build/_deps/curlpp-src/include/curlpp/Options.hpp:281:74: error: invalid conversion from 'int' to 'CURLoption' [-fpermissive]
  281 |         typedef curlpp::OptionTrait<curl_closepolicy, CURLOPT_CLOSEPOLICY> ClosePolicy;
      |                                                                          ^
      |                                                                          |
      |                                                                          int
In file included from /build/source/build/_deps/curlpp-src/src/curlpp/Easy.cpp:26:
/build/source/build/_deps/curlpp-src/include/curlpp/Options.hpp:281:74: error: invalid conversion from 'int' to 'CURLoption' [-fpermissive]
  281 |         typedef curlpp::OptionTrait<curl_closepolicy, CURLOPT_CLOSEPOLICY> ClosePolicy;
      |                                                                          ^
      |                                                                          |
      |                                                                          int
make[2]: *** [_deps/curlpp-build/CMakeFiles/curlpp.dir/build.make:160: _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Options.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [_deps/curlpp-build/CMakeFiles/curlpp.dir/build.make:76: _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Easy.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:254: _deps/curlpp-build/CMakeFiles/curlpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 23%] Linking CXX static library libpugixml.a
[ 23%] Built target pugixml
make: *** [Makefile:136: all] Error 2
```

### Additional context

Only started happening after recent flake update

### Notify maintainers
@DrymarchonShaun 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.6, NixOS, 24.11 (Vicuna), 24.11.20241105.4aa3656`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source`
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ar/arma3-unix-launcher/package.nix
+++ b/pkgs/by-name/ar/arma3-unix-launcher/package.nix
@@ -4,7 +4,6 @@
   cmake,
   curl,
   curlpp,
-  doctest,
   srcOnly,
   fetchFromGitHub,
   fetchurl,
@@ -13,7 +12,6 @@
   qt5,
   spdlog,
   substituteAll,
-  trompeloeil,
   buildDayZLauncher ? false,
 }:
 stdenv.mkDerivation (finalAttrs: {
@@ -37,7 +35,6 @@ stdenv.mkDerivation (finalAttrs: {
         rev = "45664c4e9f05ff287731a9ff8b724d0c89fb6e77";
         hash = "sha256-qLD9zD6hbItDn6ZHHWBXrAWhySvqcs40xA5+C/5Fkhw=";
       };
-      doctest_src = doctest;
       curlpp_src = srcOnly curlpp;
       fmt_src = fmt;
       nlohmann_json_src = nlohmann_json;
@@ -52,7 +49,9 @@ stdenv.mkDerivation (finalAttrs: {
         url = "https://github.com/julianxhokaxhiu/SteamworksSDKCI/releases/download/1.53/SteamworksSDK-v1.53.0_x64.zip";
         hash = "sha256-6PQGaPsaxBg/MHVWw2ynYW6LaNSrE9Rd9Q9ZLKFGPFA=";
       };
-      trompeloeil_src = trompeloeil;
+      # Only required for testing?
+      doctest_src = null;
+      trompeloeil_src = null;
     })
     # game won't launch with steam integration anyways, disable it
     ./disable_steam_integration.patch
@@ -61,14 +60,9 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [
     qt5.wrapQtAppsHook
     cmake
-  ];
-
-  buildInputs = [
     spdlog
-    curlpp.src
     curl
-    qt5.qtbase
-    qt5.qtsvg
+    curlpp
   ];
 
   cmakeFlags = [ "-Wno-dev" ] ++ lib.optionals buildDayZLauncher [ "-DBUILD_DAYZ_LAUNCHER=ON" ];

--- a/pkgs/by-name/ar/arma3-unix-launcher/package.nix
+++ b/pkgs/by-name/ar/arma3-unix-launcher/package.nix
@@ -5,6 +5,7 @@
   curl,
   curlpp,
   doctest,
+  srcOnly,
   fetchFromGitHub,
   fetchurl,
   fmt,
@@ -36,8 +37,8 @@ stdenv.mkDerivation (finalAttrs: {
         rev = "45664c4e9f05ff287731a9ff8b724d0c89fb6e77";
         hash = "sha256-qLD9zD6hbItDn6ZHHWBXrAWhySvqcs40xA5+C/5Fkhw=";
       };
-      curlpp_src = curlpp.src;
       doctest_src = doctest;
+      curlpp_src = srcOnly curlpp;
       fmt_src = fmt;
       nlohmann_json_src = nlohmann_json;
       pugixml_src = fetchFromGitHub {

